test/utils/matchers.go

```diff
@@ -2,60 +2,150 @@ package utils
 
 import (
 	"fmt"
+	"net/url"
 
+	"github.com/containers/common/pkg/config"
+	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/format"
 	"github.com/onsi/gomega/gexec"
+	"github.com/onsi/gomega/matchers"
+	"github.com/onsi/gomega/types"
 )
 
+// HaveActiveService verifies the given service is the active service
+func HaveActiveService(name interface{}) OmegaMatcher {
+	return WithTransform(
+		func(cfg *config.Config) string {
+			return cfg.Engine.ActiveService
+		},
+		Equal(name))
+}
+
+type ServiceMatcher struct {
+	types.GomegaMatcher
+	Name                  interface{}
+	URI                   interface{}
+	Identity              interface{}
+	failureMessage        string
+	negatedFailureMessage string
+}
+
+func VerifyService(name, uri, identity interface{}) OmegaMatcher {
+	return &ServiceMatcher{
+		Name:     name,
+		URI:      uri,
+		Identity: identity,
+	}
+}
+
+func (matcher *ServiceMatcher) Match(actual interface{}) (success bool, err error) {
+	cfg, ok := actual.(*config.Config)
+	if !ok {
+		return false, fmt.Errorf("ServiceMatcher matcher expects a config.Config")
+	}
+
+	if _, err = url.Parse(matcher.URI.(string)); err != nil {
+		return false, err
+	}
+
+	success, err = HaveKey(matcher.Name).Match(cfg.Engine.ServiceDestinations)
+	if !success || err != nil {
+		matcher.failureMessage = HaveKey(matcher.Name).FailureMessage(cfg.Engine.ServiceDestinations)
+		matcher.negatedFailureMessage = HaveKey(matcher.Name).NegatedFailureMessage(cfg.Engine.ServiceDestinations)
+		return
+	}
+
+	sd := cfg.Engine.ServiceDestinations[matcher.Name.(string)]
+	success, err = Equal(matcher.URI).Match(sd.URI)
```

Comment:
should we add another check here for if sd was even retrieved from the service destinations before checking for equality?